src/main/java/seedu/address/model/person/Person.java

```diff
@@ -33,9 +35,33 @@ public Person(Name name, Phone phone, Email email, Address address, Set<Tag> tag
         this.phone = phone;
         this.email = email;
         this.address = address;
+        this.owedSalary = new Money(0);
         this.tags.addAll(tags);
     }
 
+    private Person(Name name, Phone phone, Email email, Address address, Money owedSalary,
+                  Set<Tag> tags) {
+        requireAllNonNull(name, phone, email, address, owedSalary, tags);
+        this.name = name;
+        this.phone = phone;
+        this.email = email;
+        this.address = address;
+        this.owedSalary = owedSalary;
+        this.tags.addAll(tags);
+    }
+
+    /**
+     * Recalculates owed salary.
+     *
+     * @param addedSalary salary to be added.
+     * @return Person with updated salary.
+     */
+    public Person updateSalary(Money addedSalary) {
+        Money owedSalary = this.owedSalary.add(addedSalary);
+
+        return new Person(name, phone, email, address, owedSalary, tags);
+    }
+
```

Comment:
Does not address issue of subtracting `owedSalary` when job is completed.

Consider the following. Person A has done Job B that worth $10. Person A has not been paid for Job B and so `owedSalary` is $10. Soon after, Person A gets paid. A new Job B' is created which is the same as Job B except `hasPaid` is set to false. `owedSalary` should be $0. However, as the amount is not subtracted from `owedSalary`, `owedSalary` is still $10.

Do note that multiple immutable job instances may be created. Perhaps consider using the `JobList` interface to calculate the salary when it is needed.

Reply:
Keeping a record of the jobs hence seems necessary since paying a person twice for a job should not update their salary twice either; this seems to require some sort of static list to avoid duplicate instances. I was thinking of keeping track of jobIds from the Person class, then referring to some singleton Manager class to retrieve the specific jobs, idk, thoughts?

Reply:
I don't think we can pay twice for a job. All jobs are once-off, not recurring, and are like tasks in the iP. Once they are paid for, they kinda just sit around unused. They are completed

Reply:
Maybe we should have a 'history' function to see all the previous jobs of the person that have been paid for.

Reply:
For now, my current implementation has each person store a map of job ids and their paid-for status, such that the following holds:
| History      | Update to job | Result | 
| ----------- | ----------- |----------- |
| Paid   | Paid       | Nothing |
| Paid   | Not paid       | Throw exception |
| Not paid   | Paid       | Pay |
| Not-paid   | Not paid       | Nothing |
| Not in history   | Paid       | Add to history, but don't update salary |
| Not in history   | Not paid       | Add to history, update salary |